src/buffer.rs

```diff
@@ -39,102 +35,34 @@ enum Entry {
 ///
 /// *This type is available only if Syn is built with the `"parsing"` feature.*
 pub struct TokenBuffer {
-    // NOTE: Do not implement clone on this - there are raw pointers inside
-    // these entries which will be messed up. Moving the `TokenBuffer` itself is
-    // safe as the data pointed to won't be moved.
-    ptr: *const Entry,
-    len: usize,
+    // NOTE: Do not implement clone on this - while the current design could be
+    // cloned, other designs which could be desirable may not be cloneable.
+    entries: Box<[Entry]>,
 }
 
+// Keep the explicit impl for backwards compatibility.
 impl Drop for TokenBuffer {
```

Comment:
Trying to figure out the purpose of this. Is this to support code that did `fn f<T: Drop>` and called `f::<TokenBuffer>`? If so, that's "you brought this upon yourself" territory and I wouldn't care about breaking it. Is there any plausible practical code that would need this impl to exist?

Or is it the reverse concern, that there might be code that breaks if we later need to add back a Drop impl?

Reply:
It's paranoia around letter-of-law and the former — I don't think there's any other impact, since there aren't any generics to capture potentially late bound lifetimes in.